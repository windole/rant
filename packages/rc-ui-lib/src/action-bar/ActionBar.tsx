import React, { useContext, useMemo } from 'react';
import classnames from 'classnames';
import { ActionBarProps } from './PropsType';
import ActionBarContext from './ActionBarContext';
import ConfigProviderContext from '../config-provider/ConfigProviderContext';

const ActionBar: React.FC<ActionBarProps> = (props) => {
  const { prefixCls, createNamespace } = useContext(ConfigProviderContext);
  const [bem] = createNamespace('action-bar', prefixCls);

  const children = useMemo(() => React.Children.toArray(props.children), [props.children]);

  return (
    <ActionBarContext.Provider value={{ parent: { children } }}>
      <div
        className={classnames(props.className, bem(), {
          'rc-safe-area-bottom': props.safeAreaInsetBottom,
        })}
        style={props.style}
      >
        {React.Children.toArray(props.children)
          .filter(Boolean)
          .map((child: React.ReactFragment, index: number) => {
            // @ts-ignore
            return React.cloneElement(child, {
              index,
            });
          })}
      </div>
    </ActionBarContext.Provider>
  );
};

ActionBar.defaultProps = {
  safeAreaInsetBottom: true,
};

export default ActionBar;
