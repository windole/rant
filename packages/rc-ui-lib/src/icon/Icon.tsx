import React, { useContext, useMemo } from 'react';
import classnames from 'classnames';
import { addUnit } from '../utils';
import { IconProps } from './PropsType';
import Badge from '../badge';
import ConfigProviderContext from '../config-provider/ConfigProviderContext';

function isImage(name?: string): boolean {
  return name ? name.indexOf('/') !== -1 : false;
}

const Icon: React.FC<IconProps> = (props) => {
  const { iconPrefix } = useContext(ConfigProviderContext);

  const { dot, tag, name, className, onClick, onTouchStart } = props;

  const imageIcon = isImage(name);

  const classPrefix = useMemo(
    () => props.classPrefix || iconPrefix,
    [props.classPrefix, iconPrefix],
  );

  return (
    <Badge
      dot={dot}
      tag={tag}
      className={classnames(className, classPrefix, imageIcon ? '' : `${classPrefix}-${name}`)}
      style={{
        color: props.color,
        fontSize: addUnit(props.size),
        ...props.style,
      }}
      onClick={onClick}
      onTouchStart={onTouchStart}
      {...props.badge}
    >
      {props?.children}
      {imageIcon && <img className={classnames('rc-icon__image')} src={name} alt={name} />}
    </Badge>
  );
};

Icon.defaultProps = {
  tag: 'i',
};

export default Icon;
