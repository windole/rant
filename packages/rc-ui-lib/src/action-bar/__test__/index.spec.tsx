import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from '../../icon';
import ActionBar from '..';

describe('ActionBar', () => {
  let wrapper;
  afterEach(() => {
    wrapper.unmount();
    jest.restoreAllMocks();
  });

  it('should allow to disable safe-area-inset-bottom prop', () => {
    wrapper = mount(
      <ActionBar safeAreaInsetBottom={false}>
        <ActionBar.Icon icon="chat-o" text="客服" />
        <ActionBar.Icon icon="cart-o" text="购物车" />
        <ActionBar.Icon icon="shop-o" text="店铺" />
      </ActionBar>,
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('basic usage with ActionBar.Icon', async () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon="chat-o" text="客服" />
        <ActionBar.Icon icon="cart-o" text="购物车" />
        <ActionBar.Icon icon="shop-o" text="店铺" />
        <ActionBar.Button type="danger" text="立即购买" />
      </ActionBar>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when using badge prop in ActionBarIcon', async () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon="chat-o" text="客服" badge={{ dot: true }} />
        <ActionBar.Icon icon="cart-o" text="购物车" badge={{ content: 5 }} />
      </ActionBar>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when color prop in ActionBarIcon', async () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon="chat-o" color="#ee0a24" text="客服" />
        <ActionBar.Icon icon="cart-o" text="购物车" />
        <ActionBar.Icon icon="star" color="#ff5000" text="店铺" />
      </ActionBar>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when icon prop in ActionBarIcon', async () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon="chat-o" color="#ee0a24" text="客服" />
        <ActionBar.Icon icon={<Icon name="shop-o" />} text="购物车" />
      </ActionBar>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('when icon prop in ActionBarIcon is not isValidElement', async () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon={123} color="#ee0a24" text="客服" />
      </ActionBar>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when iconClass prop in ActionBarIcon', async () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon="chat-o" iconClass="icon-class" text="客服" />
        <ActionBar.Icon icon="cart-o" text="购物车" />
      </ActionBar>,
    );
    expect(wrapper.find(Icon).at(0).find('icon-class')).toBeTruthy();
  });

  it('should render correctly when iconPrefix prop in ActionBar.Icon', async () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon="chat-o" iconPrefix="my-icon" text="客服" />
        <ActionBar.Icon icon="cart-o" text="购物车" />
      </ActionBar>,
    );
    expect(wrapper.find(Icon).at(0).find('.my-icon-chat-o').exists()).toBeTruthy();
  });

  it('should emit click event in ActionBar.Icon', async () => {
    const onClick = jest.fn();
    wrapper = mount(
      <ActionBar>
        <ActionBar.Icon icon="chat-o" text="客服" onClick={onClick} />
        <ActionBar.Icon icon="cart-o" text="购物车" />
      </ActionBar>,
    );
    await wrapper.find(Icon).at(0).children('.van-icon-chat-o').first().simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('should render correctly when using type prop and text prop in ActionBarButton', () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Button type="warning" text="加入购物车" />
        <ActionBar.Button type="danger" text="立即购买" />
      </ActionBar>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when using icon prop in ActionBarButton', () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Button type="warning" text="加入购物车" icon="shop-o" />
        <ActionBar.Button type="danger" text="立即购买" icon={<Icon name="chat-o" />} />
      </ActionBar>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when using color prop in ActionBarButton', () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Button type="warning" text="加入购物车" color="red" />
        <ActionBar.Button
          type="danger"
          text="立即购买"
          color="linear-gradient(to right, #ffd01e, #ff8917)"
        />
      </ActionBar>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('when ActionBarButton does not wrapperd by ActionBar', () => {
    wrapper = mount(
      <div>
        <ActionBar.Button type="warning" text="加入购物车" />
        <ActionBar.Button type="danger" text="立即购买" />
      </div>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('when ActionBarButton has children', () => {
    wrapper = mount(
      <ActionBar>
        <ActionBar.Button type="warning">加入购物车</ActionBar.Button>
        <ActionBar.Button type="danger">立即购买</ActionBar.Button>
      </ActionBar>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
