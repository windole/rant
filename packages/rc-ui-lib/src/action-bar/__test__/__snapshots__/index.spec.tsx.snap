// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar basic usage with ActionBar.Icon 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarIcon
      icon="chat-o"
      index={0}
      key=".0"
      text="客服"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          className="rc-action-bar-icon__icon"
          name="chat-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-chat-o"
            showZero={true}
            style={
              Object {
                "color": undefined,
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-chat-o"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              }
            />
          </Badge>
        </Icon>
        客服
      </div>
    </ActionBarIcon>
    <ActionBarIcon
      icon="cart-o"
      index={1}
      key=".1"
      text="购物车"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          className="rc-action-bar-icon__icon"
          name="cart-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-cart-o"
            showZero={true}
            style={
              Object {
                "color": undefined,
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-cart-o"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              }
            />
          </Badge>
        </Icon>
        购物车
      </div>
    </ActionBarIcon>
    <ActionBarIcon
      icon="shop-o"
      index={2}
      key=".2"
      text="店铺"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          className="rc-action-bar-icon__icon"
          name="shop-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-shop-o"
            showZero={true}
            style={
              Object {
                "color": undefined,
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-shop-o"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              }
            />
          </Badge>
        </Icon>
        店铺
      </div>
    </ActionBarIcon>
    <ActionBarButton
      index={3}
      key=".3"
      text="立即购买"
      type="danger"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last rc-action-bar-button--first"
        iconPosition="left"
        size="large"
        tag="button"
        type="danger"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last rc-action-bar-button--first rc-button rc-button--danger rc-button--large"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="rc-button__content"
          >
            <span
              className="rc-button__text"
              key="text"
            >
              立即购买
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
  </div>
</ActionBar>
`;

exports[`ActionBar should allow to disable safe-area-inset-bottom prop 1`] = `"<div class=\\"rc-action-bar\\"><div role=\\"button\\" class=\\"rc-action-bar-icon\\" tabindex=\\"0\\"><div class=\\"rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-chat-o\\"></div>客服</div><div role=\\"button\\" class=\\"rc-action-bar-icon\\" tabindex=\\"0\\"><div class=\\"rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-cart-o\\"></div>购物车</div><div role=\\"button\\" class=\\"rc-action-bar-icon\\" tabindex=\\"0\\"><div class=\\"rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-shop-o\\"></div>店铺</div></div>"`;

exports[`ActionBar should render correctly when color prop in ActionBarIcon 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarIcon
      color="#ee0a24"
      icon="chat-o"
      index={0}
      key=".0"
      text="客服"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          className="rc-action-bar-icon__icon"
          color="#ee0a24"
          name="chat-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-chat-o"
            showZero={true}
            style={
              Object {
                "color": "#ee0a24",
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-chat-o"
              style={
                Object {
                  "color": "#ee0a24",
                  "fontSize": undefined,
                }
              }
            />
          </Badge>
        </Icon>
        客服
      </div>
    </ActionBarIcon>
    <ActionBarIcon
      icon="cart-o"
      index={1}
      key=".1"
      text="购物车"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          className="rc-action-bar-icon__icon"
          name="cart-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-cart-o"
            showZero={true}
            style={
              Object {
                "color": undefined,
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-cart-o"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              }
            />
          </Badge>
        </Icon>
        购物车
      </div>
    </ActionBarIcon>
    <ActionBarIcon
      color="#ff5000"
      icon="star"
      index={2}
      key=".2"
      text="店铺"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          className="rc-action-bar-icon__icon"
          color="#ff5000"
          name="star"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-star"
            showZero={true}
            style={
              Object {
                "color": "#ff5000",
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-star"
              style={
                Object {
                  "color": "#ff5000",
                  "fontSize": undefined,
                }
              }
            />
          </Badge>
        </Icon>
        店铺
      </div>
    </ActionBarIcon>
  </div>
</ActionBar>
`;

exports[`ActionBar should render correctly when icon prop in ActionBarIcon 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarIcon
      color="#ee0a24"
      icon="chat-o"
      index={0}
      key=".0"
      text="客服"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          className="rc-action-bar-icon__icon"
          color="#ee0a24"
          name="chat-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-chat-o"
            showZero={true}
            style={
              Object {
                "color": "#ee0a24",
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-chat-o"
              style={
                Object {
                  "color": "#ee0a24",
                  "fontSize": undefined,
                }
              }
            />
          </Badge>
        </Icon>
        客服
      </div>
    </ActionBarIcon>
    <ActionBarIcon
      icon={
        <Icon
          name="shop-o"
          tag="i"
        />
      }
      index={1}
      key=".1"
      text="购物车"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Badge
          className="rc-action-bar-icon__icon"
          showZero={true}
          tag="div"
        >
          <div
            className="rc-badge__wrapper rc-action-bar-icon__icon"
          >
            <Icon
              name="shop-o"
              tag="i"
            >
              <Badge
                className="van-icon van-icon-shop-o"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-shop-o"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
          </div>
        </Badge>
        购物车
      </div>
    </ActionBarIcon>
  </div>
</ActionBar>
`;

exports[`ActionBar should render correctly when using badge prop in ActionBarIcon 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarIcon
      badge={
        Object {
          "dot": true,
        }
      }
      icon="chat-o"
      index={0}
      key=".0"
      text="客服"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          badge={
            Object {
              "dot": true,
            }
          }
          className="rc-action-bar-icon__icon"
          name="chat-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-chat-o"
            dot={true}
            showZero={true}
            style={
              Object {
                "color": undefined,
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-chat-o"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              }
            >
              <div
                className="rc-badge rc-badge--dot rc-badge--fixed"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              />
            </div>
          </Badge>
        </Icon>
        客服
      </div>
    </ActionBarIcon>
    <ActionBarIcon
      badge={
        Object {
          "content": 5,
        }
      }
      icon="cart-o"
      index={1}
      key=".1"
      text="购物车"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <Icon
          badge={
            Object {
              "content": 5,
            }
          }
          className="rc-action-bar-icon__icon"
          name="cart-o"
          tag="div"
        >
          <Badge
            className="rc-action-bar-icon__icon van-icon van-icon-cart-o"
            content={5}
            showZero={true}
            style={
              Object {
                "color": undefined,
                "fontSize": undefined,
              }
            }
            tag="div"
          >
            <div
              className="rc-badge__wrapper rc-action-bar-icon__icon van-icon van-icon-cart-o"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              }
            >
              <div
                className="rc-badge rc-badge--fixed"
                style={
                  Object {
                    "background": undefined,
                  }
                }
              >
                5
              </div>
            </div>
          </Badge>
        </Icon>
        购物车
      </div>
    </ActionBarIcon>
  </div>
</ActionBar>
`;

exports[`ActionBar should render correctly when using color prop in ActionBarButton 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarButton
      color="red"
      index={0}
      key=".0"
      text="加入购物车"
      type="warning"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first"
        color="red"
        iconPosition="left"
        size="large"
        tag="button"
        type="warning"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first rc-button rc-button--warning rc-button--large"
          onClick={[Function]}
          style={
            Object {
              "background": "red",
              "borderColor": "red",
              "color": "#fff",
            }
          }
        >
          <div
            className="rc-button__content"
          >
            <span
              className="rc-button__text"
              key="text"
            >
              加入购物车
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
    <ActionBarButton
      color="linear-gradient(to right, #ffd01e, #ff8917)"
      index={1}
      key=".1"
      text="立即购买"
      type="danger"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last"
        color="linear-gradient(to right, #ffd01e, #ff8917)"
        iconPosition="left"
        size="large"
        tag="button"
        type="danger"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last rc-button rc-button--danger rc-button--large"
          onClick={[Function]}
          style={
            Object {
              "background": "linear-gradient(to right, #ffd01e, #ff8917)",
              "border": 0,
              "color": "#fff",
            }
          }
        >
          <div
            className="rc-button__content"
          >
            <span
              className="rc-button__text"
              key="text"
            >
              立即购买
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
  </div>
</ActionBar>
`;

exports[`ActionBar should render correctly when using icon prop in ActionBarButton 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarButton
      icon="shop-o"
      index={0}
      key=".0"
      text="加入购物车"
      type="warning"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first"
        icon="shop-o"
        iconPosition="left"
        size="large"
        tag="button"
        type="warning"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first rc-button rc-button--warning rc-button--large"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="rc-button__content"
          >
            <Icon
              className="rc-button__icon"
              name="shop-o"
              tag="i"
            >
              <Badge
                className="rc-button__icon van-icon van-icon-shop-o"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper rc-button__icon van-icon van-icon-shop-o"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
            <span
              className="rc-button__text"
              key="text"
            >
              加入购物车
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
    <ActionBarButton
      icon={
        <Icon
          name="chat-o"
          tag="i"
        />
      }
      index={1}
      key=".1"
      text="立即购买"
      type="danger"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last"
        icon={
          <Icon
            name="chat-o"
            tag="i"
          />
        }
        iconPosition="left"
        size="large"
        tag="button"
        type="danger"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last rc-button rc-button--danger rc-button--large"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="rc-button__content"
          >
            <Icon
              name="chat-o"
              tag="i"
            >
              <Badge
                className="van-icon van-icon-chat-o"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-chat-o"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
            <span
              className="rc-button__text"
              key="text"
            >
              立即购买
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
  </div>
</ActionBar>
`;

exports[`ActionBar should render correctly when using type prop and text prop in ActionBarButton 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarButton
      index={0}
      key=".0"
      text="加入购物车"
      type="warning"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first"
        iconPosition="left"
        size="large"
        tag="button"
        type="warning"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first rc-button rc-button--warning rc-button--large"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="rc-button__content"
          >
            <span
              className="rc-button__text"
              key="text"
            >
              加入购物车
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
    <ActionBarButton
      index={1}
      key=".1"
      text="立即购买"
      type="danger"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last"
        iconPosition="left"
        size="large"
        tag="button"
        type="danger"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last rc-button rc-button--danger rc-button--large"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="rc-button__content"
          >
            <span
              className="rc-button__text"
              key="text"
            >
              立即购买
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
  </div>
</ActionBar>
`;

exports[`ActionBar when ActionBarButton does not wrapperd by ActionBar 1`] = `
<div>
  <ActionBarButton
    text="加入购物车"
    type="warning"
  >
    <Button
      className="rc-action-bar-button rc-action-bar-button--warning"
      iconPosition="left"
      size="large"
      tag="button"
      type="warning"
    >
      <button
        className="rc-action-bar-button rc-action-bar-button--warning rc-button rc-button--warning rc-button--large"
        onClick={[Function]}
        style={Object {}}
      >
        <div
          className="rc-button__content"
        >
          <span
            className="rc-button__text"
            key="text"
          >
            加入购物车
          </span>
        </div>
      </button>
    </Button>
  </ActionBarButton>
  <ActionBarButton
    text="立即购买"
    type="danger"
  >
    <Button
      className="rc-action-bar-button rc-action-bar-button--danger"
      iconPosition="left"
      size="large"
      tag="button"
      type="danger"
    >
      <button
        className="rc-action-bar-button rc-action-bar-button--danger rc-button rc-button--danger rc-button--large"
        onClick={[Function]}
        style={Object {}}
      >
        <div
          className="rc-button__content"
        >
          <span
            className="rc-button__text"
            key="text"
          >
            立即购买
          </span>
        </div>
      </button>
    </Button>
  </ActionBarButton>
</div>
`;

exports[`ActionBar when ActionBarButton has children 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarButton
      index={0}
      key=".0"
      type="warning"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first"
        iconPosition="left"
        size="large"
        tag="button"
        type="warning"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--warning rc-action-bar-button--first rc-button rc-button--warning rc-button--large"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="rc-button__content"
          >
            <span
              className="rc-button__text"
              key="text"
            >
              加入购物车
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
    <ActionBarButton
      index={1}
      key=".1"
      type="danger"
    >
      <Button
        className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last"
        iconPosition="left"
        size="large"
        tag="button"
        type="danger"
      >
        <button
          className="rc-action-bar-button rc-action-bar-button--danger rc-action-bar-button--last rc-button rc-button--danger rc-button--large"
          onClick={[Function]}
          style={Object {}}
        >
          <div
            className="rc-button__content"
          >
            <span
              className="rc-button__text"
              key="text"
            >
              立即购买
            </span>
          </div>
        </button>
      </Button>
    </ActionBarButton>
  </div>
</ActionBar>
`;

exports[`ActionBar when icon prop in ActionBarIcon is not isValidElement 1`] = `
<ActionBar
  safeAreaInsetBottom={true}
>
  <div
    className="rc-action-bar rc-safe-area-bottom"
  >
    <ActionBarIcon
      color="#ee0a24"
      icon={123}
      index={0}
      key=".0"
      text="客服"
    >
      <div
        className="rc-action-bar-icon"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        客服
      </div>
    </ActionBarIcon>
  </div>
</ActionBar>
`;
